gpt4all-chat/modellist.cpp

```diff
@@ -319,36 +324,42 @@ bool EmbeddingModels::filterAcceptsRow(int sourceRow,
                                        const QModelIndex &sourceParent) const
 {
     QModelIndex index = sourceModel()->index(sourceRow, 0, sourceParent);
-    bool isInstalled = sourceModel()->data(index, ModelList::InstalledRole).toBool();
-    bool isEmbedding = sourceModel()->data(index, ModelList::FilenameRole).toString() == DEFAULT_EMBEDDING_MODEL ||
-        sourceModel()->data(index, ModelList::FilenameRole).toString() == NOMIC_EMBEDDING_MODEL;
-    return isInstalled && isEmbedding;
+    bool isEmbeddingModel = sourceModel()->data(index, ModelList::IsEmbeddingModelRole).toBool();
+    bool installed = sourceModel()->data(index, ModelList::InstalledRole).toBool();
+    QString filename = sourceModel()->data(index, ModelList::FilenameRole).toString();
+    auto &known = KNOWN_EMBEDDING_MODELS;
+    if (std::find(known, std::end(known), filename.toStdString()) == std::end(known)) {
+        return false; // we are currently not prepared to support other embedding models
+    }
+    return isEmbeddingModel && (!m_requireInstalled || installed);
 }
 
-int EmbeddingModels::count() const
```

Comment:
Which you're removing... what's going on?

Reply:
Not gone - just moved to the header, since its body is a single word. I could move it back if you'd like.